common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java

```diff
@@ -71,6 +76,9 @@ public FastThreadLocalThread(ThreadGroup group, Runnable target, String name, lo
      * Note that this method is for internal use only, and thus is subject to change at any time.
      */
     public final InternalThreadLocalMap threadLocalMap() {
+        if (this != Thread.currentThread() && logger.isWarnEnabled()) {
+            logger.warn("It's not thread-safe to get 'threadLocalMap' which doesn't belong to the caller thread");
```

Comment:
I wonder if we also want to add a stack trace to the log .. Same below

Reply:
It sounds reasonable, done it.